api/types/server_test.go

```diff
@@ -354,6 +357,268 @@ func TestServerCheckAndSetDefaults(t *testing.T) {
+						AWS: &AWSInfo{
+							AccountID:        "123456789012",
+							InstanceID:       "i-123456789012",
+							Region:           "us-east-1",
+							AvailabilityZone: "us-east-1b",
+							Integration:      "teleportdev",
+						},
+					},
+				},
+			},
+			assertion: func(t *testing.T, s *ServerV2, err error) {
+				require.EqualError(t, err, `AWS VPC ID is required for "openssh-ephemeral-key" SubKind`)
+			},
+		},
+		{
+			name: "OpenSSHEphemeralKey node with aws cloud metadata but missing integration",
+			server: &ServerV2{
+				Kind:    KindNode,
+				SubKind: SubKindOpenSSHEphemeralKeyNode,
+				Version: V2,
+				Metadata: Metadata{
+					Name:      "5da56852-2adb-4540-a37c-80790203f6a9",
+					Namespace: defaults.Namespace,
+				},
+				Spec: ServerSpecV2{
+					Addr:     "example:22",
+					Hostname: "openssh-node",
+					CloudMetadata: &CloudMetadata{
+						AWS: &AWSInfo{
+							AccountID:        "123456789012",
+							InstanceID:       "i-123456789012",
+							Region:           "us-east-1",
+							AvailabilityZone: "us-east-1b",
+							VPCID:            "vpc-abcd",
+						},
+					},
+				},
+			},
+			assertion: func(t *testing.T, s *ServerV2, err error) {
+				require.EqualError(t, err, `AWS OIDC Integration is required for "openssh-ephemeral-key" SubKind`)
+			},
+		},
+		{
+			name: "OpenSSHEphemeralKey node with aws cloud metadata but missing accountid",
+			server: &ServerV2{
+				Kind:    KindNode,
+				SubKind: SubKindOpenSSHEphemeralKeyNode,
+				Version: V2,
+				Metadata: Metadata{
+					Name:      "5da56852-2adb-4540-a37c-80790203f6a9",
+					Namespace: defaults.Namespace,
+				},
+				Spec: ServerSpecV2{
+					Addr:     "example:22",
+					Hostname: "openssh-node",
+					CloudMetadata: &CloudMetadata{
+						AWS: &AWSInfo{
+							AccountID:        "123456789012",
+							InstanceID:       "i-123456789012",
+							Region:           "us-east-1",
+							AvailabilityZone: "us-east-1b",
+							VPCID:            "vpc-abcd",
+							Integration:      "teleportdev",
+						},
+					},
+				},
+			},
+			assertion: func(t *testing.T, s *ServerV2, err error) {
+				require.NoError(t, err)
+				expectedServer := &ServerV2{
+					Kind:    KindNode,
+					SubKind: SubKindOpenSSHEphemeralKeyNode,
+					Version: V2,
+					Metadata: Metadata{
+						Name:      "5da56852-2adb-4540-a37c-80790203f6a9",
+						Namespace: defaults.Namespace,
+					},
+					Spec: ServerSpecV2{
+						Addr:     "example:22",
+						Hostname: "openssh-node",
+						CloudMetadata: &CloudMetadata{
+							AWS: &AWSInfo{
+								AccountID:        "123456789012",
+								InstanceID:       "i-123456789012",
+								Region:           "us-east-1",
+								AvailabilityZone: "us-east-1b",
+								VPCID:            "vpc-abcd",
+								Integration:      "teleportdev",
+							},
+						},
+					},
+				}
+				require.Equal(t, expectedServer, s)
+
+				require.True(t, s.IsOpenSSHNode(), "IsOpenSSHNode must be true for this node")
```

Comment:
```suggestion
				assert.Equal(t, expectedServer, s)
				assert.True(t, s.IsOpenSSHNode(), "IsOpenSSHNode must be true for this node")
```

https://github.com/golang/go/wiki/TestComments#keep-going

Reply:
Switched to assert